Library/Sender.ts

```diff
@@ -83,148 +92,185 @@ class Sender {
-                // This is because warn level is printed in the console by default, and we don't want to be noisy for transient and self-recovering errors
-                // Continue informing on each failure if verbose logging is being used
-                if (!this._enableDiskRetryMode || this._numConsecutiveFailures > 0 && this._numConsecutiveFailures % Sender.MAX_CONNECTION_FAILURES_BEFORE_WARN === 0) {
-                    let notice = "Ingestion endpoint could not be reached. This batch of telemetry items has been lost. Use Disk Retry Caching to enable resending of failed telemetry. Error:";
-                    if (this._enableDiskRetryMode) {
-                        notice = `Ingestion endpoint could not be reached ${this._numConsecutiveFailures} consecutive times. There may be resulting telemetry loss. Most recent error:`;
+                    // Only use warn level if retries are disabled or we've had some number of consecutive failures sending data
+                    // This is because warn level is printed in the console by default, and we don't want to be noisy for transient and self-recovering errors
+                    // Continue informing on each failure if verbose logging is being used
+                    if (!this._enableDiskRetryMode || this._numConsecutiveFailures > 0 && this._numConsecutiveFailures % Sender.MAX_CONNECTION_FAILURES_BEFORE_WARN === 0) {
+                        let notice = "Ingestion endpoint could not be reached. This batch of telemetry items has been lost. Use Disk Retry Caching to enable resending of failed telemetry. Error:";
+                        if (this._enableDiskRetryMode) {
+                            notice = `Ingestion endpoint could not be reached ${this._numConsecutiveFailures} consecutive times. There may be resulting telemetry loss. Most recent error:`;
+                        }
+                        Logging.warn(Sender.TAG, notice, error);
+                    } else {
+                        let notice = "Transient failure to reach ingestion endpoint. This batch of telemetry items will be retried. Error:";
+                        Logging.info(Sender.TAG, notice, error)
                     }
-                    Logging.warn(Sender.TAG, notice, error);
-                } else {
-                    let notice = "Transient failure to reach ingestion endpoint. This batch of telemetry items will be retried. Error:";
-                    Logging.info(Sender.TAG, notice, error)
-                }
-                this._onErrorHelper(error);
+                    this._onErrorHelper(error);
+
+                    if (typeof callback === "function") {
+                        var errorMessage = "error sending telemetry";
+                        if (error && (typeof error.toString === "function")) {
+                            errorMessage = error.toString();
+                        }
```

Comment:
On "trick" I'm using now is that to just use JSON.strinfigy() on the error -- so you still get some context.